dart/dynamics/Joint.h

```diff
@@ -119,6 +126,14 @@ class Joint
   /// Get number of generalized coordinates
   DEPRECATED(4.1) virtual size_t getDof() const = 0;
 
+  /// Get an object to access the _index-th degree of freedom (generalized
+  /// coordinate) of this Joint
+  virtual DegreeOfFreedom* getDof(size_t _index) = 0;
+
+  /// Get an object to access the _index-th degree of freedom (generalized
+  /// coordinate) of this Joint
+  virtual const DegreeOfFreedom* getDof(size_t _index) const = 0;
```

Comment:
Skeleton can get DegreeOfFreedom by the name. Shouldn't we provide the same functionality for Joint class?